modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/cache/CacheActor.scala

```diff
@@ -0,0 +1,99 @@
+package ch.epfl.bluebrain.nexus.kg.core.cache
+
+import akka.actor.{Actor, ActorLogging, PoisonPill, ReceiveTimeout}
+import akka.cluster.sharding.ShardRegion.Passivate
+import ch.epfl.bluebrain.nexus.kg.core.cache.CacheActor.Protocol._
+import ch.epfl.bluebrain.nexus.kg.core.cache.CacheError.TypeError
+import shapeless.{Typeable, the}
+
+import scala.concurrent.duration._
+
+/**
+  * Actor that stores in memory a ''Value''.
+  *
+  * @param passivationTimeout the inactivity interval after which the actor requests to be stopped
+  * @tparam Value the generic type of the value stored on this actor
+  */
+class CacheActor[Value: Typeable](passivationTimeout: FiniteDuration) extends Actor with ActorLogging {
+
+  private val Value = the[Typeable[Value]]
+
+  private var value: Option[Value] = None
+
+  override def preStart(): Unit = {
+    context.setReceiveTimeout(passivationTimeout)
+    super.preStart()
+  }
+
+  override def receive: Receive = {
+    case Get(k) =>
+      value match {
+        case Some(v) =>
+          log.debug("Cached key '{}' found with value  '{}'", k, v)
+        case None =>
+          log.debug("Cached key '{}' not found. Proceed to update pasivation timeout to 10 sec", k)
+          context.setReceiveTimeout(10 seconds)
```

Comment:
I can also change it to immediately shutdown `context.stop(self)` but I thought the customer may try several times before desisting...and so we might want to wait before getting rid of that actor.

Reply:
The problem with adjusting the receive timeout like so is that if a key gets set then the passivation is not set back to the default. And setting the receive timeout on each message is quite expensive. I think it's better to initiate the stop with `context.parent ! Passivate(stopMessage = PoisonPill)` instead of keeping the implementation like so.